### PR TITLE
INTMDB-914: HELP: Updating the name field of a mongodbatlas_project recreates a new Project

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR update the  `mongodbatlas_project` resource to use the correct API endpoint to change the name of the project.

Link to any related issue(s):  [1296](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1296), [INTMDB-220](https://jira.mongodb.org/browse/INTMDB-220), [INTMDB-914](https://jira.mongodb.org/browse/INTMDB-914)

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### Testing
I tested my changes locally by building the binary

**Plan:**
```tf
  # mongodbatlas_project.atlas-project will be updated in-place
  ~ resource "mongodbatlas_project" "atlas-project" {
        id                                               = "64a6ddcc8837b223272141f1"
      ~ name                                             = "AndreaTestProjectUpdate" -> "AndreaTest22ProjectUpdate"
        # (10 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
2023-07-06T17:31:23.861+0200 [DEBUG] command: asking for input: "\nDo you want to perform these actions?"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.
```
**Apply:**
```tf
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

### CODE REVIEW

File: website/docs/r/project.html.markdown [hunk 1]
```diff
@@ -10,8 +10,6 @@ description: |-
 
 `mongodbatlas_project` provides a Project resource. This allows project to be created.
```
Comment (Zuhairahmed) [inline website/docs/r/project.html.markdown:12]: @andreaangiolillo in docs while we are here can also update below section in docs to remove hard code of role names and link to our docs?

role_names - (Required) Each string in the array represents a project role you want to assign to the team. Every user associated with the team inherits these roles. You must specify an array even if you are only associating a single role with the team. The following are valid roles:

GROUP_OWNER
GROUP_READ_ONLY
GROUP_DATA_ACCESS_ADMIN
GROUP_DATA_ACCESS_READ_WRITE
GROUP_DATA_ACCESS_READ_ONLY
GROUP_CLUSTER_MANAGER

**Review by Zuhairahmed — APPROVED** [review]
Minor comment, not blocking. Long over due fix, customers are going to love it! LGTM

File: website/docs/r/project.html.markdown (hunk 1 above)
Comment (andreaangiolillo) [reply website/docs/r/project.html.markdown:12]: Updated

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by martinstibbe — APPROVED** [review]
LGTM

**Pull request merged**